psasim/test/Makefile

```diff
@@ -1,12 +1,22 @@
 INCLUDE := -I../include/ -I./psa_manifest
+LIB := -L../src -lpsaff
+
+TEST_BIN = 	psa_client \
+			psa_partition
 
 .PHONY: all clean
 
-all:
-	psa_autogen manifest.json
-	$(CC) psa_ff_bootstrap_TEST_PARTITION.c -lpsaff -o partition
-	$(CC) client.c -lpsaff -o client
+all: $(TEST_BIN)
+
+psa_client:
+	$(CC) $(INCLUDE) $(CFLAGS) client.c $(LIB) -o $@
```

Comment:
This is missing a dependency when run the first time

```
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C test CFLAGS="-Wall -Werror -std=c99 -D_XOPEN_SOURCE=1 -D_POSIX_C_SOURCE=200809L"
cc -I../include/ -I./psa_manifest -Wall -Werror -std=c99 -D_XOPEN_SOURCE=1 -D_POSIX_C_SOURCE=200809L client.c -L../src -lpsaff -o psa_client
client.c:10:10: fatal error: 'psa_manifest/sid.h' file not found
#include "psa_manifest/sid.h"
         ^~~~~~~~~~~~~~~~~~~~
```

So something like this is needed

```
diff --git a/psasim/test/Makefile b/psasim/test/Makefile
index b5044b5..4b9d023 100644
--- a/psasim/test/Makefile
+++ b/psasim/test/Makefile
@@ -8,13 +8,15 @@ TEST_BIN = 	psa_client \
 
 all: $(TEST_BIN)
 
+psa_client: ../psa_manifest/sid.h
+
 psa_client:
 	$(CC) $(INCLUDE) $(CFLAGS) client.c $(LIB) -o $@
 
 psa_partition: psa_ff_bootstrap_TEST_PARTITION.c
 	$(CC) $(INCLUDE) $(CFLAGS) $< $(LIB) -o $@
 
-psa_ff_bootstrap_%.c:
+../psa_manifest/sid.h psa_ff_bootstrap_%.c:
 	../tools/psa_autogen.py manifest.json
 
 clean:
```

Reply:
Oh, I launched "make run" several times now, but this never happened to me. To me this looks like a make order, isn't it? Just out of my curiosity: do you have any idea of why this never happened to me?

Reply:
I assume at some point you got `sid.h` created; my tree didn't have this until I added the dependency